packages/ui/components/CodeEditor.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { useState, useEffect } from 'react';
+import debounce from 'lodash/debounce';
+
+
+interface CodeEditorProps {
+  schema: string;
+  onSchemaChange: (newSchema: string) => void;
+}
+
+export const CodeEditor: React.FC<CodeEditorProps> = ({ schema, onSchemaChange }) => {
```

Comment:
```suggestion
interface CodeEditorProps {
  schema: SchemaObject;
  onChange: (newSchema: SchemaObject) => void;
}

export const CodeEditor: React.FC<CodeEditorProps> = ({ schema, onChange }) => {
```

A couple of things:
1. You have the SchemaObject type [here](https://github.com/asyncapi/parser-js/blob/master/src/spec-types/v3.ts#L401) and [here](https://github.com/asyncapi/parser-js/blob/master/src/spec-types/v2.ts#L227). You may want to propose that we extract them in a separate package so you don't depend on parser here.
2. I'd get rid of `Schema` in `onSchemaChange` because it's redundant.

Reply:
@fmvilas Could you explain this further. Is it like extracting a `SchemaObject` from the `parser` into a separate file and using this as a type instead of using its string type?

Reply:
Yes, that's exactly what I meant 👍